ledger/storage/payload_storage.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/hash"
+)
+
+var _ ledger.PayloadStorage = (*PayloadStorage)(nil)
+
+type PayloadStorage struct {
```

Comment:
PayloadStorage implements the payload storage interface. 

It wraps the inner key-value storage layer to handle encoding-decoding of the payload, so that the inner storage module can simply store bytes, and the upper layer (mtrie) can pass in the payload struct to be stored.